site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -1,62 +1,155 @@
-import { shallowEqual, useActor, useMachine, useSelector } from "@xstate/react"
+import { shallowEqual, useActor, useSelector } from "@xstate/react"
+import {
+  checkAuthorization,
+  createWorkspace,
+  getTemplates,
+  getTemplateVersionSchema,
+  getWorkspaceQuota,
+} from "api/api"
 import { FeatureNames } from "api/types"
+import { ParameterSchema, Template, User, WorkspaceQuota } from "api/typesGenerated"
 import { useOrganizationId } from "hooks/useOrganizationId"
-import { FC, useContext } from "react"
+import { FC, useContext, useEffect, useState } from "react"
 import { Helmet } from "react-helmet-async"
 import { useNavigate, useParams } from "react-router-dom"
 import { pageTitle } from "util/page"
-import { createWorkspaceMachine } from "xServices/createWorkspace/createWorkspaceXService"
 import { selectFeatureVisibility } from "xServices/entitlements/entitlementsSelectors"
 import { XServiceContext } from "xServices/StateContext"
 import { CreateWorkspaceErrors, CreateWorkspacePageView } from "./CreateWorkspacePageView"
 
 const CreateWorkspacePage: FC = () => {
-  const xServices = useContext(XServiceContext)
   const organizationId = useOrganizationId()
   const { template } = useParams()
   const templateName = template ? template : ""
   const navigate = useNavigate()
+  const xServices = useContext(XServiceContext)
+  const [authState] = useActor(xServices.authXService)
+  const { me } = authState.context
   const featureVisibility = useSelector(
     xServices.entitlementsXService,
     selectFeatureVisibility,
     shallowEqual,
   )
   const workspaceQuotaEnabled = featureVisibility[FeatureNames.WorkspaceQuota]
 
-  const [authState] = useActor(xServices.authXService)
-  const { me } = authState.context
-  const [createWorkspaceState, send] = useMachine(createWorkspaceMachine, {
-    context: { organizationId, templateName, workspaceQuotaEnabled, owner: me ?? null },
-    actions: {
-      onCreateWorkspace: (_, event) => {
-        navigate(`/@${event.data.owner_name}/${event.data.name}`)
+  const [owner, setOwner] = useState<User | null>(me ?? null)
+  const [templates, setTemplates] = useState<Template[]>()
+  const [selectedTemplate, setSelectedTemplate] = useState<Template>()
+  const [templateSchema, setTemplateSchema] = useState<ParameterSchema[]>()
+  const [creatingWorkspace, setCreatingWorkspace] = useState<boolean>(false)
+  const [createWorkspaceError, setCreateWorkspaceError] = useState<Error | unknown>()
+  const [getTemplatesError, setGetTemplatesError] = useState<Error | unknown>()
+  const [getTemplateSchemaError, setGetTemplateSchemaError] = useState<Error | unknown>()
+  const [permissions, setPermissions] = useState<Record<string, boolean>>()
+  const [checkPermissionsError, setCheckPermissionsError] = useState<Error | unknown>()
+  const [workspaceQuota, setWorkspaceQuota] = useState<WorkspaceQuota>()
+  const [getWorkspaceQuotaError, setGetWorkspaceQuotaError] = useState<Error | unknown>()
```

Comment:
I'm not sure if all of these need to be stateful. For example, we likely don't need a `useState` for `getTemplatesError`. Instead, we might have a generic stateful variable, `error`, that we hydrate with errors that come back from the API. 

To further simplify, you could look into [`useReducer`](https://reactjs.org/docs/hooks-reference.html#usereducer) or the context API. 

`useReducer` is like `useState` on steroids and is something I reach for if I find myself with a bunch of `useState`s.

The [context API](https://reactjs.org/docs/context.html#contextprovider) allows one to inject contexts (objects) to any component wrapped within a Provider.  In this case, you could lean on two contexts: `workspaceContext` and `templateContext`. Each might have properties like `workspace` or `template`, `loading`, and `error`.